mne/io/eeglab/eeglab.py

```diff
@@ -445,3 +471,45 @@ def __init__(self, input_fname, events=None, event_id=None, tmin=0,
             reject=reject, flat=flat, reject_tmin=reject_tmin,
             reject_tmax=reject_tmax, add_eeg_ref=False, verbose=verbose)
         logger.info('Ready.')
+
+
+def _events_from_eeglab_raw(fname, event_id=dict()):
+    """Create events array from EEGLAB structure by looking them up in the
+    event_id, trying to reduce them to their integer part otherwise, and
+    entirely dropping them (with a warning) if this is impossible"""
+    from scipy.io import loadmat
+
+    eeg = loadmat(fname, struct_as_record=False, squeeze_me=True)["EEG"]
+    types = [event.type for event in eeg.event]
+    latencies = [event.latency for event in eeg.event]
+
+    not_in_event_id = set([x for x in types if x not in event_id])
+    not_purely_numeric = set([x for x in not_in_event_id if not x.isdigit()])
+    no_numbers = set([x for x in not_purely_numeric
+                      if not any([d.isdigit() for d in x])])
+    have_integers = set([x for x in not_purely_numeric
+                         if x not in no_numbers])
+    if len(not_purely_numeric) > 0:
+        basewarn = "Events like the following will be "
+        n_no_numbers, n_have_integers = len(no_numbers), len(have_integers)
+        if n_no_numbers > 0:
+            nonumwarm = "dropped entirely: {}, {} in total"
+            warnings.warn(basewarn + nonumwarm.format(list(no_numbers)[:5],
+                                                      n_no_numbers))
+        if n_have_integers > 0:
+            intwarn = "reduced to their integer part: {}, {} in total"
+            warnings.warn(basewarn + intwarn.format(list(have_integers)[:5],
+                                                    n_have_integers))
+        warnings.warn("Use the `event_id` keyword to "
+                      "include such events manually.")
+
+    events = list()
+    for t, latency in zip(types, latencies):
+        try:
+            event_code = event_id.get(t, int("".join([x for x in t
```

Comment:
Yup, but that's not a problem of the event reader incorrectly reading events, but of how we handle it returning nothing later on.
(Possibly, it should return a single row of 3 zeros if it finds no events.)

I think it would be the best plan to fix the (lots of) problems after there is a way of reading in events for `preload=False` - if my current code here is extremely bad, maybe everything else would have to be done differently.